src/value/owned.rs

```diff
@@ -606,6 +609,22 @@ mod value_tests {
                 assert_eq!(Some(&vec![("greetings", OwnedElement::from(OwnedValue::String("hello".into())))].into_iter().collect()), e.as_struct());
             }
         ),
+        case::struct_with_no_text(
+            OwnedValue::Struct(vec![(local_sid_token(21).with_source("hello_table", 2), OwnedValue::String("hello".into()))].into_iter().collect()),
+            IonType::Struct,
+            AsStruct,
+            &|e: &OwnedElement| {
+                assert_eq!(Some(&vec![(local_sid_token(21).with_source("hello_table", 2), OwnedValue::String("hello".into()))].into_iter().collect()), e.as_struct());
+            }
+        ),
+        case::struct_with_no_text(
+            OwnedValue::Struct(vec![(local_sid_token(21), OwnedValue::String("hello".into()))].into_iter().collect()),
+            IonType::Struct,
+            AsStruct,
+            &|e: &OwnedElement| {
+                assert_eq!(Some(&vec![(local_sid_token(21), OwnedValue::String("hello".into()))].into_iter().collect()), e.as_struct());
+            }
+        ),
```

Comment:
I think this test is too trivial, consider the following:

```rust
        case::struct_with_no_text(
            OwnedValue::Struct(
                vec![
                    (local_sid_token(21), OwnedValue::String("hello".into())),
                    (local_sid_token(22), OwnedValue::String("world".into())),
                ].into_iter().collect()),
            IonType::Struct,
            AsStruct,
            &|e: &OwnedElement| {
                assert_eq!(
                    Some(&vec![
                        (local_sid_token(22), OwnedValue::String("world".into())),
                        (local_sid_token(21), OwnedValue::String("hello".into())),
                    ].into_iter().collect()),
                    e.as_struct()
                );
            }
        ),
```

I am pretty sure this would fail, but if I order the tokens the same it would succeed:

```rust
        case::struct_with_no_text(
            OwnedValue::Struct(
                vec![
                    (local_sid_token(21), OwnedValue::String("hello".into())),
                    (local_sid_token(22), OwnedValue::String("world".into())),
                ].into_iter().collect()),
            IonType::Struct,
            AsStruct,
            &|e: &OwnedElement| {
                assert_eq!(
                    Some(&vec![
                        (local_sid_token(21), OwnedValue::String("hello".into())),
                        (local_sid_token(22), OwnedValue::String("world".into())),
                    ].into_iter().collect()),
                    e.as_struct()
                );
            }
        ),
```

What I am doing here is testing something akin to `{$20: "hello", $21: "world"} == {$21: "world", $20: "hello"}`.

Reply:
I also am suspicious of this modified case for named fields:

```rust
        case::struct_(
            OwnedValue::Struct(
                vec![
                    ("greetings", OwnedElement::from(OwnedValue::String("hello".into()))),
                    ("greetings", OwnedElement::from(OwnedValue::String("world".into()))),
                ].into_iter().collect()
            ),
            IonType::Struct,
            AsStruct,
            &|e: &OwnedElement| {
                assert_eq!(
                    Some(
                        &vec![
                            ("greetings", OwnedElement::from(OwnedValue::String("world".into()))),
                            ("greetings", OwnedElement::from(OwnedValue::String("hello".into()))),
                        ].into_iter().collect()
                    ),
                    e.as_struct()
                );
            }
        ),
```

When I pulled this code down this also did not work.  I think you will need to do something similar for the fields with text as my suggested fix for the `no_text_fields`.